### PR TITLE
Ct 1138 dont create new brach every run

### PR DESCRIPTION
**Pull request by romanbracinik**
Jira: CT-1138
KBC: XXX

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

Tag presiel [tu](https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=39296&view=results)  komplet zeleny.

Changes:
- typo
- zacal som reusovat branche
- unikatne nazvy bucketov pre paratesty

### CODE REVIEW

File: tests/ClientProvider/TestSetupHelper.php [hunk 1]
```diff
@@ -56,8 +56,12 @@ public function setUpForProtectedDevBranch(
             $branchName = $clientProvider->getDevBranchName();
             // dev can create & delete branches in production
             $devBranches = new DevBranches($clientProvider->getDefaultClient(['token' => STORAGE_API_DEVELOPER_TOKEN]));
-            $this->deleteBranchesByPrefix($devBranches, $branchName);
-            $branch = $devBranches->createBranch($branchName);
+
+            $branch = $this->getExistingBranches($devBranches, $branchName);
+
+            if ($branch === null) {
+                $branch = $devBranches->createBranch($branchName);
```
Comment (romanbracinik) [inline tests/ClientProvider/TestSetupHelper.php:63]: vytvaram novu branch len ked neexistuje pre dany test, vsade kde sa to pouziva sa nejakou formou premazavaju buckty/metadata a pod. takze test prechadza a myslim si e to pomaha na err You don't have access to resource.

File: tests/Backend/CommonPart1/BucketsTest.php [hunk 2]
```diff
@@ -194,12 +194,7 @@ public function testBucketEvents(string $devBranchType, string $userRole): void
         $this->assertEventWithRetries($this->_testClient, $assertCallback, $query);
 
         $bucketEvents = $this->_testClient->listBucketEvents($bucketId);
-        if ($devBranchType === ClientProvider::DEFAULT_BRANCH) {
-            // there are already events in production
-            $this->assertGreaterThan(2, $bucketEvents);
-        } else {
-            $this->assertCount(2, $bucketEvents);
-        }
+        $this->assertGreaterThan(2, $bucketEvents);
```
Comment (romanbracinik) [inline tests/Backend/CommonPart1/BucketsTest.php:197]: tim ze sa pouziva stale jedna branch a nevytvara nova tak je to ako v maine a tych eventov tam moze byt viac

File: tests/Backend/CommonPart1/BucketsTest.php [hunk 3]
```diff
@@ -90,8 +90,8 @@ public function testBucketsList(string $devBranchType, string $userRole): void
      */
     public function testBucketDetail(string $devBranchType, string $userRole): void
     {
+        $bucketName = sha1($this->generateDescriptionForTestObject()).'-testBucketDetail';
```
[the remark this answers is not in the document]
Comment (jirkasemmler) [reply tests/Backend/CommonPart1/BucketsTest.php:93]: to co delas by melo udelat i getTestBucketId() ne? Jen teda nechapu v cem toto ma pomoci

File: tests/ClientProvider/TestSetupHelper.php [hunk 4]
```diff
@@ -110,4 +114,20 @@ public function provideComponentsClientTypeBasedOnSuite(StorageApiTestCase $that
...
+    private function getExistingBranches(DevBranches $devBranches, string $branchPrefix): ?array
+    {
+        $branchesList = $devBranches->listBranches();
+        $branchesCreatedByThisTestMethod = array_filter(
+            $branchesList,
+            function ($branch) use ($branchPrefix) {
+                return strpos($branch['name'], $branchPrefix) === 0;
+            },
+        );
+        foreach ($branchesCreatedByThisTestMethod as $branch) {
```
Comment (jirkasemmler) [inline tests/ClientProvider/TestSetupHelper.php:127]: `return reset(branchesCreatedByThisTestMethod) ?: null;`

**Review by jirkasemmler — CHANGES_REQUESTED** [review]

File: tests/Backend/CommonPart1/BucketsTest.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (romanbracinik) [reply tests/Backend/CommonPart1/BucketsTest.php]: Hmm no jasne, neviem asi som to v tom ako som to rozne skusal upravil stim, ze to ma nejaky vyznam a pak som zil v tej domienke ze to tak ma byt:) Dik zmazal som to.

**Comment by romanbracinik** [comment]
Dik upravil som to a davam na review @jirkasemmler

File: tests/ClientProvider/TestSetupHelper.php [hunk 5]
```diff
@@ -110,4 +109,23 @@ public function provideComponentsClientTypeBasedOnSuite(StorageApiTestCase $that
...
+    {
+        $branchesList = $devBranches->listBranches();
+        $branchesCreatedByThisTestMethod = array_filter(
+            $branchesList,
+            function ($branch) use ($branchName) {
+                return $branch['name'] === $branchName;
+            },
+        );
+
+        $branch = reset($branchesCreatedByThisTestMethod) ?: null;
```
Comment (jirkasemmler) [inline tests/ClientProvider/TestSetupHelper.php:123]:
```suggestion
        $branch = reset($branchesCreatedByThisTestMethod);
```

File: tests/ClientProvider/TestSetupHelper.php [hunk 6]
```diff
@@ -110,4 +109,23 @@ public function provideComponentsClientTypeBasedOnSuite(StorageApiTestCase $that
...
+
+        if ($branch === null) {
```
Comment (jirkasemmler) [inline tests/ClientProvider/TestSetupHelper.php:125]:
```suggestion
        if ($branch === false) {
```

File: tests/ClientProvider/TestSetupHelper.php (hunk 5 above)
Comment (jirkasemmler) [reply tests/ClientProvider/TestSetupHelper.php:123]: ten ternarni operator jsem tam daval jen protoze `reset` vraci `false` pokud je pole prazdne

**Review by jirkasemmler — CHANGES_REQUESTED** [review]
jen malickost.

**Review by jirkasemmler — APPROVED** [review]

**Pull request merged**